### PR TITLE
Handle CLEAR_ALL and EVICT_ALL events for query cache

### PR DESCRIPTION
**Pull request by ahmetmircik**
- [ ] backport 

closes https://github.com/hazelcast/hazelcast/issues/12599

### CODE REVIEW

**Comment by ahmetmircik** [comment]
verify

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/SubscriberAccumulatorHandler.java [hunk 1]
```diff
@@ -57,13 +66,28 @@ public void handle(QueryCacheEventData eventData, boolean ignored) {
             case EVICTED:
                 queryCache.deleteInternal(keyData, false, entryEventType);
                 break;
-            // TODO: if we want strongly consistent clear & evict, removal can be made based on sequence and partition ID
             case CLEAR_ALL:
+                handleMapWideEvent(eventData, entryEventType, evictAllCount, evictedEntryCount);
```
Comment (taburet) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/SubscriberAccumulatorHandler.java]: Seems like this line should be swapped with the line in the next `case`.

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/SubscriberAccumulatorHandler.java]: good catch, fixed

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/DefaultQueryCache.java [hunk 2]
```diff
@@ -498,6 +485,23 @@ public Indexes getIndexes() {
...
+    @Override
+    public int removeEntriesOf(int partitionId) {
+        int removedEntryCount = 0;
+
+        Set<Data> keys = recordStore.keySet();
+        Iterator<Data> iterator = keys.iterator();
+        while (iterator.hasNext()) {
+            Data keyData = iterator.next();
+            if (context.getPartitionId(keyData) == partitionId) {
+                recordStore.remove(keyData);
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/DefaultQueryCache.java]: I think this should read
```
if (recordStore.remove(keyData) != null) {
  removedEntryCount++;
}
```

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/DefaultQueryCache.java]: done

**Pull request merged**